Tests/SQLKitTests/SQLCommonTableExpressionTests.swift

```diff
@@ -0,0 +1,243 @@
+import SQLKit
+import XCTest
+
+final class SQLCommonTableExpressionTests: XCTestCase {
+    var db = TestDatabase()
+
+    override class func setUp() {
+        XCTAssert(isLoggingConfigured)
+    }
+    
+    func testSelectQueryWithCTE() {
+        XCTAssertSerialization(
+            of: self.db.select().with("a", columns: ["b"], as: SQLSubquery.select { $0.column(SQLLiteral.numeric("1")) }).column(SQLColumn("b", table: "a")),
+            is: "WITH ``a`` (``b``) AS (SELECT 1) SELECT ``a``.``b``"
+        )
+
+        XCTAssertSerialization(
+            of: self.db.select().with(SQLIdentifier("a"), columns: ["b"], as: SQLSubquery.select { $0.column(SQLLiteral.numeric("1")) }).column(SQLColumn("b", table: "a")),
+            is: "WITH ``a`` (``b``) AS (SELECT 1) SELECT ``a``.``b``"
+        )
+
+        XCTAssertSerialization(
+            of: self.db.select().with("a", columns: [SQLIdentifier("b")], as: SQLSubquery.select { $0.column(SQLLiteral.numeric("1")) }).column(SQLColumn("b", table: "a")),
+            is: "WITH ``a`` (``b``) AS (SELECT 1) SELECT ``a``.``b``"
```

Comment:
Can we use this API without the `(``b``)`?
If yes, please add some tests :)
If not, why the limitation?

Reply:
If you have to ask, you didn't look at the API very closely at all 🤨